rbac/internal/views.py

```diff
@@ -403,20 +403,27 @@ def invalid_default_admin_groups(request):
     return HttpResponse('Invalid method, only "DELETE" and "GET" are allowed.', status=405)
 
 
-def role_removal(request, role_uuid):
+def role_removal(request):
     """View method for internal role removal requests.
 
-    DELETE /_private/api/utils/roles/<role_uuid>/
+    DELETE /_private/api/utils/roles/
     """
     logger.info(f"Role removal: {request.method} {request.user.username}")
     if request.method == "DELETE":
         if not destructive_ok():
             return HttpResponse("Destructive operations disallowed.", status=400)
 
-        role_obj = get_object_or_404(Role, uuid=role_uuid)
+        role_name = request.GET.get("name")
+        if not role_name:
+            return HttpResponse(
+                'Invalid request, must supply the "name" query parameter.',
+                status=400,
+            )
+        # Add system true to prevent deletion of custom roles
+        role_obj = get_object_or_404(Role, name=role_name, system=True)
```

Comment:
can we specify here also tenant_id for certitude ?

Reply:
I was talking to Keith yesterday, this system flag is redundant now since all system=True must be public tenant. So I just choose the system.

Reply:
Let me switch to tenant since I created a ticket to remove this flag in the future